guardian/shortcuts.py

```diff
@@ -404,15 +408,18 @@ def get_objects_for_user(user, perms, klass=None, use_groups=True, any_perm=Fals
         group_model = get_group_obj_perms_model(queryset.model)
         group_filters = {
             'permission__content_type': ctype,
-            'permission__codename__in': codenames,
             'group__%s' % get_user_model().groups.field.related_query_name(): user,
         }
+        if len(codenames):
+            group_filters.update({
+                'permission__codename__in': codenames,
+            })
```

Comment:
Similar here. Just splitting filter building into two steps, where one is conditional.